### DISCUSSION TITLE
Zigbee2Tasmota doesn't seem to like what it receives back from certain Ledvance bulb types.

### DISCUSSION THREAD

**Discussion by brivett** [arendst/Tasmota#16507] [Q&A]
So, I've been having much fun linking a variety of Zigbee devices to my existing openHab system.  Zigbee2Tasmota outputting MQTT is a great match as it separates the uncertainty or the Zigbee side from the unpredictability of openHab (which I love as a system)

I'm using a SONOFF Zigbee Pro bridge, flashed with Tasmota 12.1.0.1

Program Version | 12.1.0.1(zbbrdgpro)
2022-08-20T11:49:26
2_0_4_1/v4.4.3

I have integrated Osram power sockets, Ledvance E27 bulbs, a Ikea Tradfri LED system , some temp and humidity sensors and wall plate dimmer switches from Samotech and Candeo.  Everything works.   Until...

I've added an initial 4 GU10 RGBW LEDs model AC33906 by LEDVANCE

They do exactly what I tell them.... Change colour, dim, turn off etc....   The only issue is that sometimes the information fed back from the LED to the coordinator is not correctly received or decoded.

Thus, I might tell all bulbs (individual commands not groups) to go green.  They do this, but they may then report back this change, but only the status of 3 out of four bulbs gets correctly decoded.  thus, tasmota is confused and so is openHab as it's fed duff information.

In the log I will see lots of entries similar to :

22:40:18.590 ZbInput discarding byte 45
22:40:18.594 ZbInput discarding byte C4
22:40:18.596 ZbInput discarding byte 40
22:40:18.598 ZbInput discarding byte 51
22:40:18.602 ZbInput discarding byte 01
22:40:18.605 ZbInput discarding byte C0
22:40:18.608 ZbInput discarding byte F5
22:40:18.611 ZbInput discarding byte A1

Sometimes preceded by a problem message for a ZbRecieved statement.

One short level 3 log is attached - It's very repeatable so I can created whatever might be needed.

Has anyone seen similar to this before?  Should I just stop trying to get feedback from the LEDs?

Any help is much appreciated.

Barry

[weblog.txt](https://github.com/arendst/Tasmota/files/9561106/weblog.txt)

**Comment by s-hadinger** [comment]
This is super weird. I have never seen such messages except at boot. It means that the Start of Frame `0xFE` was not read correctly.

Since failed messages are not linked to a specific zigbee command, the only explanation I can think of is electrical interferences: changes in the bulbs might generate spiked on the power supply. How far is the ZBBridge Pro from the bulbs? Are they on the same plugs?

**Reply by brivett** [reply]
Hi again

So my test Sonoff Zigbee Bridge Pro arrived (well two actually)

I had no problem building my original one, and I have no problem building either of these, I just can't make them work with Zigbee.

I've tried a base install then expand disks, transfer files and start that way
I've tried using the facory all included bin
I've even ised ESPtool to take a backup of my working unit, and then flashed them with that....

Which ever method I use a see a similar log (attached.)

Always ends :

20:12:30.562 MQT: tele/zbpro_01/RESULT = {"ZbState":{"Status":99,"Message":"Abort"}}
20:12:30.566 ZIG: Abort
20:12:30.568 ZIG: Stopping (99)

...and Zigbee is not started  :-)

Any ideas ?

Thanks
Barry

[ZigbeeBridgeWontStart.txt](https://github.com/arendst/Tasmota/files/9610691/ZigbeeBridgeWontStart.txt)

**Comment by brivett** [comment]
Thanks for coming back to me on this.  Your efforts are greatly appreciated.

Here's what I know.
It's a domestic house.
The Zigbee Bridge Pro is in an upstairs room, the lights are in the ceiling of a room which is across a hallway.
The Lights are on a standard lighting circuit which has no other zigbee devices on it.
There is no shortage of repeated Zigbee signal due to using power sockets which repeat (Also OSRAM/LEDvance)
There are other colour swtching LEDvance lights of another model around the house - no problems.
The problem can be encouraged by repeated switching of the colours via commands.  If there is only one light being commanded it is less likely to happen. Commanding 4 lights in immediate succession makes it far more likely to happen.
If I move the lights to a wall socket based circuit < 1m from the Zigbee Bridge Pro, it still happens.
If I create a group of the 4 lights (so they are all switched and all report at the same moment) it happens pretty much every time.
The problem can cause the whole Zigbee Bridge Pro's zigbee operation to crash/stop.


I have a second Zigbee Bridge Pro on order to use as a test bed - the first one is firmly in place running stuff - and is great, when the.

MY background is as an electronics engineer, turned application software specialist... not a coder but can clunge scripts together.

Ever seen a zigbee map do as per the attached ?

https://user-images.githubusercontent.com/47494323/190987589-1be33516-354e-4a9d-87e5-9bfc24826620.mp4

Thanks
Barry

**Reply by s-hadinger** [reply]
Wow, nice map.

**Reply by sfromis** [reply]
You're not the first to get "child birthday crazy party" when the network map gets so crowded that the usual rules for pushing device distance starts to resonate instead of settling.

**Reply by brivett** [reply]
So I don't worry about that - It doesn't seem to bother anything... I think....

**Reply by sfromis** [reply]
The JavaScript engine in the browser is getting a bit of a workout, but probably not slowing things much down.

**Comment by brivett** [comment]
Hi again

So my test Sonoff Zigbee Bridge Pro arrived (well two actually)

I had no problem building my original one, and I have no problem building either of these, I just can't make them work with Zigbee.

I've tried a base install then expand disks, transfer files and start that way
I've tried using the facory all included bin
I've even ised ESPtool to take a backup of my working unit, and then flashed them with that....

Which ever method I use a see a similar log (attached.)

Always ends :

20:12:30.562 MQT: tele/zbpro_01/RESULT = {"ZbState":{"Status":99,"Message":"Abort"}}
20:12:30.566 ZIG: Abort
20:12:30.568 ZIG: Stopping (99)

...and Zigbee is not started  :-)

Any ideas ?

Thanks
Barry
[ZigbeeBridgeWontStart.txt](https://github.com/arendst/Tasmota/files/9610693/ZigbeeBridgeWontStart.txt)

**Reply by sfromis** [reply]
You did run the procedure to update the firmware in the CC2652 Zigbee chip?

**Reply by s-hadinger** [reply]
The easy way is to flash the specific firmware with Tasmota Web Installer. It contains everything needed. 

What firmware did you flash on the CC2652P?

**Reply by brivett** [reply]
Is this the process where we use Berry console to :

import sonoff_zb_pro_flasher as cc
cc.load("SonoffZBPro_coord_20220219.hex")
cc.check()

(Which responded Verification of hex file OK)

Then :
cc.flash()

Or have I missed something...  Zigbee not started

The log during what I think is the flashing of the CC2652

22:05:40.210 HTP: Berry Console
22:05:44.688 FLH: Starting verification of HEX file
22:06:01.958 FLH: Verification of HEX file OK
22:06:24.159 FLH: cc2652_flasher rx=23 tx=19 rst=15 bsl=22
22:06:24.233 FLH: Flashing started (takes 5 minutes during which Tasmota is unresponsive)
22:11:27.242 FLH: Flashing completed: OK
22:11:27.355 FLH: Flash crc32 0x000000 - 0x2FFFF = bytes('1598929A')
22:11:27.395 MQT: Attempting connection...
22:11:27.438 MQT: Connected
22:11:27.445 MQT: tele/zbpro_01/LWT = Online (retained)
22:11:27.452 MQT: cmnd/zbpro_01/POWER = 
22:11:27.456 MQT: Subscribe to cmnd/zbpro_01/#
22:11:27.461 MQT: Subscribe to cmnd/tasmotas/#
22:11:27.465 MQT: Subscribe to cmnd/ZBPRO-01_fb/#
22:11:27.570 HTP: Main Menu
22:11:28.437 MQT: Attempting connection...
22:11:28.471 MQT: Connected
22:11:28.479 MQT: tele/zbpro_01/LWT = Online (retained)
22:11:28.485 MQT: cmnd/zbpro_01/POWER = 
22:11:28.489 MQT: Subscribe to cmnd/zbpro_01/#
22:11:28.493 MQT: Subscribe to cmnd/tasmotas/#
22:11:28.497 MQT: Subscribe to cmnd/ZBPRO-01_fb/#
22:11:41.685 HTP: Main Menu
22:11:49.849 HTP: Consoles
22:11:51.082 HTP: Console

**Comment by brivett** [comment]
Some success and some head scratching - need to write up what I learned so will revert to you (hopefully tomorrow)

The key issue was building more than one directly from the factory.bin with everything ready made devices which had the same PanID

More later :-)

Thanks

**Reply by sfromis** [reply]
The expectation would be that you'd not get same PanID, unless you did it yourself. Docs says:
> At initial start a pseudo-random PanID is derived from the ESP MAC address.

**Reply by brivett** [reply]
Hi

OK, so here is an update on what I learned.

A lot.  I have a far better understanding of the Zigbee side of Z2T than before.  Some aspects are still a puzzle, but the fog is clearing.

So, as said, I had ordered two additional ZB Pro's, one direct from China and one from within the UK.  They both arrived within hours of each other.

My plan was to take a copy of my working ZB Pro and thus create an exact clone of it to experiment with.  it would of course have a different MAC address and IP address and the MQTT client names etc could be adjusted.

So, I read the flash from the working one and tried to push it into one of the other units - Yes the clone unit ran, I could see my settings, but Zigbee did not start.  (I'll call that test a)

So I flashed in the factory.bin version and again yes, it ran, virginal but zigbee would not start  (test b)

I went round and round for hours until I went back and read the documentation - specifically about PanID's

What that told me was that I can't have two Zigbee networks in proximaty which have the same PanID.  So, in test a where I created a close of the working unit, there would I assume then have been two units with identical PanIDs, which I believe would have stopped the new one from starting.   I suspect that had I switched off the working one, the new clone would have started ?

So, I circled round, trying more times than I can remember, and I am 99.9% certain that I saw using the factory.bin flash method, two devices assume the same PanID....

Anyway, in the end I managed to create three working units, each with unique PanID's   Two for live use and one for test.

Which leads to another question...

All three devices have the same H & L keys - Thoughts on that ?   Does that mean that anyone who creates their Zigbee network via the factory.bin software will have the same keys too ?

My actual goal is to have a method where I can create a clone unit to test upgrades etc on... what is the recommended way to transfer all setting and knowledge of devices to a test unit (whilst switching off the live one for PanID reasons) ?

As for the starting point, errant LEDvance GU10's which Z2T doesn't receieve messages correctly from....

I now have those 4 LED's on a separate Z2T bridge on their own channel away from WiFi in use, close to the bridge... and they behave just the same.  I note from LEDvance that there is new firmware for these LEDs....  Are there any options for feeding Z2T the firmware so they can OTA update... or are there other methods ?

Thank you so much....

**Reply by sfromis** [reply]
As I've understood it (not strong on details), the PanID is stored in the Zigbee chip (as it to id of the coordinator as such, which is the Zigbee chip), and messing around with Tasmota will not replace it, unless you take specific action to start a new network.

Z2T does not include the feature of upgrading end device firmware, you could temporarily pair with Z2M for that. At least the older non-"pro" Sonoff Zigbee bridge should work with Z2M too for such operations, in a similar manner as connecting to ZHA.

My preference for device management firmly excludes ideas about cloning devices as a good mode of operation, I prefer to know specifically what I've done to set up a device.

**Reply by brivett** [reply]
The clone option was to enable a backup device to be held in store.   With over 60 Sonoff units, I have real experience of their reliability, so I try and keep a spare of everything that could be needed...  The kit is very cheap, time is expensive.  I re-paired nearly 30 devices yesterday after rebuilding the units.  It has led me to believe that Zigbee has very specific pros and cons.   Great where battery is the only option e.g. Temperature and humidity sensors, great where  you don't want to run a cable up a wall to a switch (My Mystrom WifI buttons are all now replaced with Zigbee scene controllers), and the no neutral Zigbee based mains dimmer switches are excellent... the WiFi equivalents require a neutral.  

Given the potential hassle of re-pairing after failure, I'm very glad to be running Z2T as it partitions these devices from my openHab system that would be far more hassle to correct.   I plugged the Z2T bridge into a USB battery pack and carried it around the building to re-pair rather than rounding up the devices....

I guess I'm going to have to learn Z2M for OTA upgrades.. I have Tuya hub but that denies that anything is ever upgradable....   :-)

**Reply by s-hadinger** [reply]
@brivett Since you copied only the last 3 lines, I assumed that the Zigbee MCU did not start at all, and I didn't check the file you attached. Now looking at the detailed log, the Zigbee MCU did start but did not enable the coordinator mode.

The EZSP MCU code first checks if the PanID is not already in use on the radio network. If the PanID it wants to use is already present, it refuses to start to avoid a major conflict. This is why you cannot have 2 coordinators with the same PanID.

Z2T does not support OTA of zigbee devices, it's a bit too complex for poor ESP; although some courageous people could probably do it in Berry.

I need to try randomization of H/L keys and PanID. I'm surprised that your devices chose the same keys.

**Reply by brivett** [reply]
Thanks Stephan - I assume it is not possible to change the keys once items are paired....

**Reply by s-hadinger** [reply]
Changing the H/L keys needs a repairing, and if you do so, I suggest you also change the PanID so the device doesn't try to reconnect with obsolete keys.
